projects/storefrontstyles/scss/_features.scss

```diff
@@ -0,0 +1,62 @@
+$featureMessagingItems: true;
+
+// Prefix for feature's CSS classes that are added to the root element
+//
+// The same as the one used in the TS service FeatureStylesService
+$CSS_FEATURE_FLAG_PREFIX: 'cxFeat_';
+
+// Returns a feature flag's CSS class name.
+//
+// The CSS class name consists of the prefix `cxFeat_`
+// and the feature flag name.
+@function getFeatureCssClass($feature) {
+  @return $CSS_FEATURE_FLAG_PREFIX + $feature;
+}
+
+// Applies styles activated when a specific feature flag is enabled in the Spartacus global config.
+//
+// IMPORTANT: Please mind to call `useFeatureStyles(feature)` from `@spartacus/core`
+//            in the constructor of your styled component.
+//            Otherwise the feature-flagged styles MIGHT NOT be activated
+//            EVEN if the flag is enabled in the Spartacus configuration!
+//
+//            For instance, if the SCSS of your component you have:
+//            ```scss
+//            %cx-my-component {
+//              ...
+//                ...
+//                  @include useFeatureStyles('myFeatureFlag') { ... }
+//            }
+//            ```
+//            ...then you should also call `useFeatureStyles('myFeatureFlag')` in the constructor of this component.
+//
+// Use case: Breaking style changes can be added for a specific feature flag
+// and customers can only use those styles if they explicitly enable a feature flag.
+//
+// The mixin requires only a $feature name as argument.
+//
+// The second argument with value `'currentSelector'` should be passed ONLY
+// in one edge case, when the feature-flagged styles are defined
+// for the Angular root element, i.e. `cx-storefront`.
+@mixin useFeatureStyles($feature, $classTarget: 'ancestorSelector') {
```

Comment:
I would not use `useFeatureStyles` here. For me `useFeatureStyles` means 'I want to register a feature flag for evaluation' while here it is about a check whether the flag is actually active. Also think about an additional mixin that checks if the flag is _not_ active (btw do you agree that we need that as well? But we could also use an additional parameter then).

So I think the initial `forFeature` was not that bad...

Reply:
1.
Thank you for feedback. Reverted mixin name to `forFeature`.

2.
> Also think about an additional mixin that checks if the flag is not active (btw do you agree that we need that as well? But we could also use an additional parameter then).
I'm not convinced yet we'd need an opposite mixin. Do you have any specific usecase in mind?
I imagine the normal styles should be defined normally, but only new breaking changes styles should be wrapped within the extra selector.

Theoretically we could add a negative variant of the mixin which under the hood would use `:not()`, but it may impact negatively the CSS performance and in extreme case make the browser slow down on user's interactions.
Sharing snippet just for reference:
```scss
@mixin notForFeature($feature) {
  $featureClass: getFeatureCssClass($feature);
  @at-root :not(.#{$featureClass}) & {
    @content;
  }
}
```
If you have other ideas of implementing a notForFeature, that doesn't use `:not()`, please let me know,

Reply:
Here I was wrong and confused view changes (where we frequently worked with a negative variant) with SCSS changes. I confirmed with Larisa, who mostly takes care of our styling: On the SCSS side, we should bei fine with this additive approach.